modules/cf-core/src/cfCore.ts

```diff
@@ -313,36 +312,31 @@ export class CFCore {
             break;
           }
 
-          case PersistStateChannelType.SyncNumProposedApps: {
-            await this.storeService.incrementNumProposedApps(stateChannel.multisigAddress);
-            break;
-          }
-
-          case PersistStateChannelType.SyncRejectedProposal: {
-            await this.storeService.removeAppProposal(
-              stateChannel.multisigAddress,
-              appContext.identityHash,
-              stateChannel.toJson(),
+          case PersistStateChannelType.SyncRejectedProposals: {
+            // NOTE: this relies on `removeAppProposal` being idempotent
+            // and functional concurrent writes of the store
```

Comment:
We could, but we have concurrent store tests so I think seeing if this works, then if it fails seeing if making that changes fixes it. But i dont see any reason a failure here (unless its in the store action itself) would be a problem since rejections are processed at the next sync regardless